File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

bool compare(string a, string b){
    return a+b > b+a;
}

string solution(vector<int> numbers) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    vector<string> tmp;
    string answer = "";
    
    for(auto num :numbers)
        tmp.push_back(to_string(num));
    sort(tmp.begin(), tmp.end(), compare);
    for(auto a:tmp)
        answer += a;
    if(answer[0] == '0') answer = "0";
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    new_numbers = sorted(map(str, numbers),key= lambda x: x*3, reverse=True) # 문자열로 변경 -> 3자리 붙인 기준 내림차순 정렬
    answer = ''.join(new_numbers)
    return answer if answer[0] != '0' else '0'